nwcsafpps_runner/tests/test_pps_hook.py

```diff
@@ -84,13 +88,64 @@
         posttroll_topic: "/PPSv2018"
 """
 
+TEST_YAML_NAMESERVERS_CONTENT_OK = """
+pps_hook:
+    post_hook: !!python/object:nwcsafpps_runner.pps_posttroll_hook.PPSMessage
+      description: "This is a pps post hook for PostTroll messaging"
+      metadata:
+        station: "norrkoping"
+        output_format: "CF"
+        level: "2"
+        variant: DR
+        geo_or_polar: "polar"
+        software: "NWCSAF-PPSv2018"
+        nameservers:
+        - 'test1'
+        - 'test2'
+"""
+
+TEST_YAML_NAMESERVERS_IS_LIST = """
+pps_hook:
+    post_hook: !!python/object:nwcsafpps_runner.pps_posttroll_hook.PPSMessage
+      description: "This is a pps post hook for PostTroll messaging"
+      metadata:
+        station: "norrkoping"
+        output_format: "CF"
+        level: "2"
+        variant: DR
+        geo_or_polar: "polar"
+        software: "NWCSAF-PPSv2018"
+        nameservers: test1
+"""
+
 
 def create_instance_from_yaml(yaml_content_str):
     """Create a PPSMessage instance from a yaml file."""
     from nwcsafpps_runner.pps_posttroll_hook import PPSMessage
     return yaml.load(yaml_content_str, Loader=yaml.UnsafeLoader)
 
 
+class TestPPSPublisher(unittest.TestCase):
+    """Test the PPSPublisher."""
+
+    def setUp(self):
+        self.test_nameservers = ['test1', 'test2']
+
+    def test_called_with_nameserver(self):
+        """Test calling the PPSPublisher with a list of specified nameservers."""
+        mymock = MagicMock()
+
+        manager = Manager()
+        publisher_q = manager.Queue()
+
+        with patch('nwcsafpps_runner.pps_posttroll_hook.Publish', return_value=mymock) as mypatch:
+            pub_thread = PPSPublisher(publisher_q, self.test_nameservers)
+            pub_thread.start()
+
+        mypatch.assert_called_with('PPS', 0, nameservers=self.test_nameservers)
+        pub_thread.stop()
```

Comment:
Can you make one test with no nameserver specified?

Reply:
Sure, sorry, of course! Added